miio/waterheater.py

```diff
@@ -0,0 +1,219 @@
+        {'washStatus': 1, 'velocity': 0, 'waterTemp': 29,
+        'targetTemp': 70, 'errStatus': 0, 'hotWater': 60,
+        'needClean': 0, 'modeType': 1, 'appointStart': 7
+        'appointEnd': 0}
+        """
+        self.data = data
+
+    @property
+    def status(self) -> OperationStatus:
+        return OperationStatus(self.data["washStatus"])
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is currently on."""
+        return not self.status == OperationStatus.TurnedOff
+
+    @property
+    def velocity(self) -> int:
+        return self.data["velocity"]
+
+    @property
+    def water_temperature(self) -> int:
+        """Current water temperature."""
+        return self.data["waterTemp"]
+
+    @property
+    def target_temperature(self) -> int:
+        """Target water temperature."""
+        return self.data["targetTemp"]
+
+    @property
+    def error(self) -> int:
+        return self.data["errStatus"]
+
+    @property
+    def hot_water_volume(self) -> int:
+        return self.data["hotWater"]
+
+    @property
+    def cleaning_required(self) -> int:
+        return self.data["needClean"]
+
+    @property
+    def mode(self) -> OperationMode:
+        return OperationMode(self.data["modeType"])
+
+    @property
+    def booking_time_start(self) -> int:
+        return self.data["appointStart"]
+
+    @property
+    def booking_time_end(self) -> int:
+        return self.data["appointEnd"]
+
+class WaterHeater(Device):
+    """Main class representing the Waterheater."""
+
+    @command(
+        default_output=format_output(
+            "",
+            "Operation status: {result.status.name} ({result.status.value})\n"
+            "Velocity: {result.velocity}\n"
+            "Water temperature: {result.water_temperature}°C\n"
+            "Target temperature: {result.target_temperature}°C\n"
+            "Error status: {result.error}\n"
+            "Remaining hot water volume: {result.hot_water_volume}%\n"
+            "Device cleaning is required: {result.cleaning_required}\n"
+            "Operation mode type: {result.mode.name} ({result.mode.value}) \n"
+            "Operation start time in booking mode: {result.booking_time_start}\n"
+            "Operation end time in booking mode: {result.booking_time_end}\n",
+        )
+    )
+    def status(self) -> WaterHeaterStatus:
+        """Retrieve properties."""
+        properties = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["available_properties"]
+
+        # A single request is limited to 16 properties. Therefore the
+        # properties are divided into multiple requests
+        _props_per_request = 15
+
+        # The E1 is limited to a single property per request
+        if self.model in [MODEL_WATERHEATER_E1]:
+            _props_per_request = 1
+
+        values = self.get_properties(properties, max_properties=_props_per_request)
```

Comment:
```suggestion
        values = self.get_properties(properties, max_properties=1)
```
I would just simplify this. If there will be another waterheater based on the same series that supports multi-queries, we can adapt this afterwards.